src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -607,21 +660,85 @@ else if (assignDepth >= 1) {
+     *
+     * @param ast to check if surrounded by unnecessary parentheses
+     * @return true if Unnecessary parenthesis
+     */
+    private static boolean checkConditionalOrRelationalOperator(DetailAST ast) {
         final int type = ast.getType();
         final int parentType = ast.getParent().getType();
-        final boolean isConditional = TokenUtil.isOfType(type, CONDITIONALS_AND_RELATIONAL);
-        boolean result = TokenUtil.isOfType(parentType, CONDITIONALS_AND_RELATIONAL);
-        if (isConditional) {
-            if (type == TokenTypes.LOR) {
-                result = result && !TokenUtil.isOfType(parentType, TokenTypes.LAND);
-            }
-            result = result && !TokenUtil.isOfType(parentType, TokenTypes.EQUAL,
-                TokenTypes.NOT_EQUAL);
+        final boolean isParentEqualityOperator =
+                TokenUtil.isOfType(parentType, TokenTypes.EQUAL, TokenTypes.NOT_EQUAL);
+        final boolean result;
+        if (type == TokenTypes.LOR) {
+            result = !TokenUtil.isOfType(parentType, TokenTypes.LAND)
+                    && !TokenUtil.isOfType(parentType, BITWISE_BINARY_OPERATORS);
+        }
+        else if (type == TokenTypes.LAND) {
+            result = !TokenUtil.isOfType(parentType, BITWISE_BINARY_OPERATORS);
         }
         else {
-            result = result && TokenUtil.isOfType(type, UNARY_AND_POSTFIX);
+            result = true;
         }
-        return result;
+        return result && !isParentEqualityOperator
+                && isBitWiseBinaryOrConditionalOrRelationalOperator(parentType);
+    }
+
+    /**
+     * Check if Bitwise binary operator has unnecessary parentheses.
+     *
+     * @param ast to check if surrounded by unnecessary parentheses
+     * @return true if Unnecessary parenthesis
+     */
+    private static boolean checkBitwiseBinaryOperator(DetailAST ast) {
+        final int type = ast.getType();
+        final int parentType = ast.getParent().getType();
+        final boolean result;
+        if (type == TokenTypes.BOR) {
+            result = !TokenUtil.isOfType(parentType, TokenTypes.BAND, TokenTypes.BXOR)
+                    && !TokenUtil.isOfType(parentType, RELATIONAL_OPERATOR);
+        }
+        else if (type == TokenTypes.BXOR) {
+            result = !TokenUtil.isOfType(parentType, TokenTypes.BAND)
+                    && !TokenUtil.isOfType(parentType, RELATIONAL_OPERATOR);
+        }
+        // we deal with bitwise AND here.
+        else {
+            result = !TokenUtil.isOfType(parentType, RELATIONAL_OPERATOR);
+        }
+        return result && isBitWiseBinaryOrConditionalOrRelationalOperator(parentType);
+    }
+
+    /**
+     * Check if token type is BitWise Binary Or Conditional or Relational Operator.
+     *
+     * @param type Token type to check
+     * @return true if it is BitWise Binary Or Conditional Operator
+     */
+    private static boolean isBitWiseBinaryOrConditionalOrRelationalOperator(int type) {
```

Comment:
`BitWise` => `Bitwise`

You did this casing in other areas.

Reply:
done